crates/solidity/src/solc/solc_compiler.rs

```diff
@@ -52,46 +64,73 @@ impl Compiler for SolcCompiler {
         allow_paths: Option<String>,
     ) -> anyhow::Result<StandardJsonOutput> {
         let version = self.version()?;
+        trace!("In standard_json");
 
         let mut command = std::process::Command::new(self.executable.as_str());
         command.stdin(std::process::Stdio::piped());
         command.stdout(std::process::Stdio::piped());
-        command.arg("--standard-json");
 
-        if let Some(base_path) = base_path {
-            command.arg("--base-path");
-            command.arg(base_path);
-        }
-        for include_path in include_paths.into_iter() {
-            command.arg("--include-path");
-            command.arg(include_path);
+        if let Some(base_path) = &base_path {
+            if SUPPORTS_BASE_PATH.matches(&version.default) {
+                command.arg("--base-path").arg(base_path);
+
+                if SUPPORTS_INCLUDE_PATH.matches(&version.default) {
+                    // `--base-path` and `--include-path` conflict if set to the same path, so
+                    // as a precaution, we ensure here that the `--base-path` is not also used
+                    // for `--include-path`
+                    for path in include_paths.iter().filter(|p| *p != base_path) {
+                        command.arg("--include-path").arg(Path::new(path));
+                    }
+                }
+            }
+            command.current_dir(base_path);
         }
-        if let Some(allow_paths) = allow_paths {
-            command.arg("--allow-paths");
-            command.arg(allow_paths);
+
+        if let Some(path) = &allow_paths {
+            if !path.is_empty() {
+                command.arg("--allow-paths");
+                command.arg(path);
+            }
         }
+        command.arg("--standard-json");
 
         input.normalize(&version.default);
-
-        let suppressed_warnings = input.suppressed_warnings.take().unwrap_or_default();
+        trace!("Normalized input: {:#?}", input);
 
         let input_json = serde_json::to_vec(&input).expect("Always valid");
+        trace!("Input JSON length: {}", input_json.len());
+        trace!(
+            "Input JSON content: {}",
+            String::from_utf8_lossy(&input_json)
+        );
 
-        let process = command.spawn().map_err(|error| {
+        let mut process = command.spawn().map_err(|error| {
             anyhow::anyhow!("{} subprocess spawning error: {:?}", self.executable, error)
         })?;
-        process
-            .stdin
-            .as_ref()
-            .ok_or_else(|| anyhow::anyhow!("{} stdin getting error", self.executable))?
-            .write_all(input_json.as_slice())
-            .map_err(|error| {
+
+        let suppressed_warnings = input.suppressed_warnings.take().unwrap_or_default();
```

Comment:
Please keep in sync changes in 
 `solc_compiler.rs` with https://github.com/paritytech/revive/blob/main/crates/solidity/src/solc/soljson_compiler.rs

Reply:
Explain @smiasojed

Reply:
First of all you can build revive as native binary or wasm blob, which you can use in browsers.
solc_compiler is used in native compiler binary and soljson_compiler in wasm version of compiler. If you introduce change to solc_compiler only, it means that resolc binary will not generate the same output as resolc.wasm compiler version

Reply:
Alright but when you said above `Please keep in sync changes in` what does this mean?

Reply:
I mean that any changes made to `solc_compiler.rs` should also be reflected in [`soljson_compiler.rs`](https://github.com/paritytech/revive/blob/main/crates/solidity/src/solc/soljson_compiler.rs)

Reply:
@smiasojed please check again